# Introduction

### Introduction or week 1 work

After opening the file the 1st thing i noticed it that we need to manage the upper portion of the bok having title and oher things etc.

In [1]:
f = open("PeterPan.txt", "r", encoding="utf-8-sig")
t = f.read()
len(t)

274951

The charater length of the text in the book is 274951 which means there are total 274951 characters in the whole book.

# NLTK TOKANIZERS

## Split the book to sentences

In [2]:
import nltk.tokenize as tok
sents = tok.sent_tokenize(t, language= "english")
len(sents)

2563

Yes the number corrosponds to the number of my expectations.

In [3]:
print(sents[0])
print ("------------------")
print(sents[1])
print ("------------------")


The Project Gutenberg eBook of Peter Pan, by James M. Barrie

This eBook is for the use of anyone anywhere in the United States and
most other parts of the world at no cost and with almost no restrictions
whatsoever.
------------------
You may copy it, give it away or re-use it under the terms
of the Project Gutenberg License included with this eBook or online at
www.gutenberg.org.
------------------


In [4]:
import re
t = re.sub(r"\n+", " ", t) #re.sub() is subsitude method in regular expression.
print(t[:100])

The Project Gutenberg eBook of Peter Pan, by James M. Barrie This eBook is for the use of anyone any


In [6]:
tokens = tok.word_tokenize(t, language="english")
len(tokens)

61946

In [17]:
f = open("peterpan.txt", "r", encoding="utf-8-sig")
t = f.read()
tokens = tok.word_tokenize(t, language="english")
print(len(t))
print(tokens[:30])


274951
['The', 'Project', 'Gutenberg', 'eBook', 'of', 'Peter', 'Pan', ',', 'by', 'James', 'M.', 'Barrie', 'This', 'eBook', 'is', 'for', 'the', 'use', 'of', 'anyone', 'anywhere', 'in', 'the', 'United', 'States', 'and', 'most', 'other', 'parts', 'of']


# Frequency distribution and pandas

In [24]:
import collections
c = collections.Counter(tokens) # The Counter class is used to create a dictionary-like object 
del c["and"]
print(c.most_common(10))
print(c["with"])

[('the', 2329), ('“', 1464), ('”', 1464), ('to', 1222), ('a', 959), ('of', 923), ('was', 923), ('he', 878), ('in', 698), ('it', 661)]
361


In [10]:
import string
string.punctuation

'!"#$%&\'()*+,-./:;<=>?@[\\]^_`{|}~'

In [15]:
def remove_punctuation(text):
    translator = str.maketrans("", "", string.punctuation)
    return text.translate(translator)
clean_text= remove_punctuation(t)

In [25]:
len(clean_text)
tokens = tok.word_tokenize(clean_text, language="english")
c = collections.Counter(tokens) # The Counter class is used to create a dictionary-like object 
print(c.most_common(10))
print(c["with"])

[('the', 2329), ('“', 1464), ('”', 1464), ('and', 1420), ('to', 1222), ('a', 959), ('of', 923), ('was', 923), ('he', 878), ('in', 698)]
361


In [30]:
import pandas as pd
data = c.most_common(10)
print (data)
df= pd.DataFrame(data, columns= ["Word","Frequency"])
print(df)

[('the', 2329), ('“', 1464), ('”', 1464), ('and', 1420), ('to', 1222), ('a', 959), ('of', 923), ('was', 923), ('he', 878), ('in', 698)]
  Word  Frequency
0  the       2329
1    “       1464
2    ”       1464
3  and       1420
4   to       1222
5    a        959
6   of        923
7  was        923
8   he        878
9   in        698


In [35]:
tokens = tok.word_tokenize(clean_text, language="english")
print("Tokens= ", len(tokens))
print("Type of tokens= ", len(set(tokens)))
print("Ratio= ",len(tokens)/len(set(tokens)))

Tokens=  54354
Type of tokens=  5836
Ratio=  9.313570938999314
